Code/GraphMol/FMCS/MaximumCommonSubgraph.cpp

```diff
@@ -456,15 +530,24 @@ bool MaximumCommonSubgraph::growSeeds() {
     {
       const Seed& fs = Seeds.front();
       // bigger substructure found
-      if (fs.CopyComplete)
+      if (fs.CopyComplete) {
+        bool possibleMCS = false;
         if ((!Parameters.MaximizeBonds &&
              (fs.getNumAtoms() > getMaxNumberAtoms() ||
               (fs.getNumAtoms() == getMaxNumberAtoms() &&
-               fs.getNumBonds() > getMaxNumberBonds()))) ||
-            (Parameters.MaximizeBonds &&
-             (fs.getNumBonds() > getMaxNumberBonds() ||
-              (fs.getNumBonds() == getMaxNumberBonds() &&
-               fs.getNumAtoms() > getMaxNumberAtoms())))) {
+               fs.getNumBonds() > getMaxNumberBonds())))) {
+          possibleMCS = true;
+        } else if (Parameters.MaximizeBonds &&
+                   (fs.getNumBonds() > getMaxNumberBonds() ||
+                    (fs.getNumBonds() == getMaxNumberBonds() &&
+                     fs.getNumAtoms() > getMaxNumberAtoms()))) {
+          possibleMCS = true;
+        }
+        // #945: test here to see if the MCS actually has all rings closed
+        if (possibleMCS && Parameters.BondCompareParameters.CompleteRingsOnly) {
+          possibleMCS = checkIfRingsAreClosed(fs);
```

Comment:
We know the MCS is a match between the structures, here just check to see if rings are closed in the match, if not, keep going.